File: app/backend/src/main/java/at/maturaballmanager/model/Company.java
package at.maturaballmanager.model;

import at.maturaballmanager.services.ValidationTool;
import com.fasterxml.jackson.annotation.JsonIgnore;
import io.quarkus.hibernate.orm.panache.PanacheEntity;
import jakarta.persistence.*;

import java.io.Serializable;
import java.util.Set;

@Entity
public class Company extends PanacheEntity implements Serializable {

    @Column(length = 50)
    public String name;

    @OneToMany(mappedBy = "company")
    public Set<Bill> bills;

    @OneToOne
    @JoinTable(name = "FK_ContactPerson_Address")
    public Address address;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinTable(name = "FK_ContactPerson_Company")
    public ContactPerson contactPerson;

    @Column(length = 100, nullable = false)
    public String website;

    @Column(length = 50)
    @JsonIgnore
    public String imagePath;

    public Company(String name, Address address, String website) {
        this.name = name;
        this.address = address;
        this.website = website;
    }

    public Company() {

    }

    @PrePersist
    @PreUpdate
    private void validate() {
        this.website = ValidationTool.validateWebsite(website);
    }
}

File: app/backend/src/main/kotlin/at/htlleonding/maturaballmanager/model/entities/Company.kt
package at.htlleonding.maturaballmanager.model.entities

import at.htlleonding.maturaballmanager.model.Address
import io.quarkus.hibernate.reactive.panache.PanacheEntityBase
import jakarta.persistence.*
import jakarta.validation.constraints.*

@Entity
@Table(name = "companies")
class Company: PanacheEntityBase() {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: String? = null

    @NotBlank(message = "Name ist erforderlich")
    @Size(max = 100, message = "Name darf maximal 100 Zeichen lang sein")
    var name: String? = null

    @NotBlank(message = "Branche ist erforderlich")
    @Size(max = 100, message = "Branche darf maximal 100 Zeichen lang sein")
    var industry: String? = null

    @Pattern(
        regexp = "^$|(https?://)?([\\da-z.-]+)\\.([a-z.]{2,6})([/\\w .-]*)*/?",
        message = "Bitte eine gültige URL eingeben"
    )
    var website: String? = null

    @Email(message = "Bitte eine gültige E-Mail-Adresse eingeben")
    @Column(name = "office_email")
    var officeEmail: String? = null

    @Pattern(
        regexp = "^$|^[+]?[(]?[0-9]{1,4}[)]?[-\\s./0-9]*$",
        message = "Bitte eine gültige Telefonnummer eingeben"
    )
    @Column(name = "office_phone")
    var officePhone: String? = null

    @Embedded
    var address: Address? = null

    @OneToMany(mappedBy = "company", cascade = [CascadeType.ALL], fetch = FetchType.EAGER, targetEntity = ContactPerson::class)
    var contactPersons: List<ContactPerson> = mutableListOf()

    @ManyToOne(cascade = [CascadeType.ALL])
    @JoinColumn(name = "prom_id")
    var prom: Prom? = null

}
